### PR TITLE
Add implementation comments for key rotation.

### LINKED ISSUE #290: Key Update and time to next possible update
So Section 5.3 in -06 says: 

`When this specification is used, endpoints SHOULD wait for at least three times the largest PTO among all the paths before initiating a new key update after receiving an acknowledgement that confirms receipt of the previous key update. This interval is different from that of QUIC version 1 which used three times the PTO of the only one active path.`

After having done some testing with key update in our implementations with different length paths some conclusions have been drawn.

First is that it difficult for an endpoint to know when an old key can safely be dropped. With multiple path being possible to send on, and not necessarily used as well as to send ACKs on. Thus the fact that one get key phase back as a responder indicate that beyond on-path reordering things should be settled. But for multipath one can conclude for paths being used that key-phase changes have happened. But for path currently not receiving any packet one don't know if they are just being significantly delayed or may never have been sent. Also the path RTT samples one have might not be relevant for the current path delay. Thus, beyond observing that one get some key-phase change back, and starting a really long timer it is not obvious that you can generate an algorithm that enables one to know that no outstanding old keys are there. 

The fact that this time needs to be longer than worst case RTT changes on each path do result that using 3*PTO is likely to short to re-enable key-update. At least unless one want to require trial decryption of the packets. This as keeping the old key while using the current key, so when one see a key-phase different one can try if it is the old key, if that fails and the next key works, then one drop the old key, and from that point assume this key-phase will be the new key. And thus initiating a key_update.

### PR DESCRIPTION
**Pull request by huitema**
Close #290

### CODE REVIEW

File: draft-ietf-quic-multipath.md [hunk 1]
```diff
@@ -1038,6 +1038,23 @@ adopt a simple coordination rule, such as only letting the client
...
+do not retain old read keys for no more than three times the PTO after
+having received a packet protected using the new keys. The main difference
+is that endpoints using the multipath extensions consider the largest
+of the path RTO, instead of only considering the RTO of a single path.
+
+The choice of a 3*RTO delay is a trade-off.
+Longer delays would diminish the probability that packets will be
+lost if they arrive after
+their decryption key has been discarded, but
+keeping old keys available for longer delays reduce the security of the protocol.
```
Comment (Yanmei-Liu) [inline draft-ietf-quic-multipath.md:1055]:
```suggestion
keeping old keys available for longer delays reduces the security of the protocol.
```

**Review by Yanmei-Liu — APPROVED** [review]
Looks good! Thanks for the guidance.

**Comment by mirjak** [comment]
You added this section in the implementation guidance. But I don't think you actually provide any additional implementation guidance, or? You rather provide some reasoning why 3x the largest RTO is specified and what the trade-off is. I think this would fit fine into existing Key Update section (section 4.2). More concretely, I think you can remove the first summary paragraph and simply more the second paragraph into the existing Key Update section. Or do I miss something on additional implementation guidance here?

**Comment by huitema** [comment]
Will try moving the text as suggested.

**Comment by huitema** [comment]
@mirjak Can you review again please? I think I followed your suggestion

File: draft-ietf-quic-multipath.md [hunk 2]
```diff
@@ -1038,7 +1047,6 @@ adopt a simple coordination rule, such as only letting the client
 initiate closure of duplicate paths, or perhaps relying on
 the application protocol to decide which paths should be closed.
```
Comment (mirjak) [inline draft-ietf-quic-multipath.md:1049]:
```suggestion


```

**Review by qdeconinck — APPROVED** [review]

**Pull request merged**